vars/cloudFoundryDeploy.groovy

```diff
@@ -285,11 +303,40 @@ def deployMta(config) {
         }
     }
 
-    def deployStatement = "cf ${deployCommand} ${config.mtaPath} ${config.mtaDeployParameters} ${config.mtaExtensionDescriptor}"
+    def deployStatement = "cf ${deployCommand} ${config.mtaPath} ${config.mtaDeployParameters} ${mtaExtensionDescriptorParam}"
     def apiStatement = "cf api ${config.cloudFoundry.apiEndpoint} ${config.apiParameters}"
 
-    echo "[${STEP_NAME}] Deploying MTA (${config.mtaPath}) with following parameters: ${config.mtaExtensionDescriptor} ${config.mtaDeployParameters}"
-    deploy(apiStatement, deployStatement, config, null)
+    echo "[${STEP_NAME}] Deploying MTA (${config.mtaPath}) with following parameters: ${mtaExtensionDescriptorParam} ${config.mtaDeployParameters}"
+    try {
+        deploy(apiStatement, deployStatement, config, null)
+    } finally {
+        if (config.mtaExtensionCredentials && config.mtaExtensionDescriptor && fileExists(config.mtaExtensionDescriptor)) {
+            sh "mv --force ${config.mtaExtensionDescriptor}.original ${config.mtaExtensionDescriptor} || echo 'The file ${config.mtaExtensionDescriptor}.original could not be renamed.'"
+        }
+    }
+}
+
+private void handleMtaExtensionCredentials(Map<?, ?> config) {
+    echo "[${STEP_NAME}] Modifying ${config.mtaExtensionDescriptor}. Adding credential values from Jenkins."
```

Comment:
I would vote for doing nothing in case no mtaExtensionCredentials  are defined

Reply:
The function is only called in case mtaExtensionCredentials are defined: https://github.com/SAP/jenkins-library/blob/mtaextension-cred/vars/cloudFoundryDeploy.groovy#L293
If I am not missing anything then nothing happens if no credentials are specified?